src/WebJobs.Extensions.DurableTask/Listener/TaskNonexistentActivityShim.cs

```diff
@@ -0,0 +1,37 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.Threading.Tasks;
+using DurableTask.Core;
+using DurableTask.Core.Common;
+using DurableTask.Core.Exceptions;
+
+namespace Microsoft.Azure.WebJobs.Extensions.DurableTask.Listener
+{
+    internal class TaskNonexistentActivityShim : TaskActivity
+    {
+        private readonly DurableTaskExtension config;
+        private readonly string activityName;
+        private readonly string message;
```

Comment:
We probably don't need to store this message, as we can derive it from `activityName` and keep the message closer to where we throw the exception.